src/Symfony/Component/Console/Descriptor/JsonDescriptor.php

```diff
@@ -152,15 +152,14 @@ private function getInputDefinitionData(InputDefinition $definition)
      */
     private function getCommandData(Command $command)
     {
-        $command->getSynopsis();
         $command->mergeApplicationDefinition(false);
 
         return array(
             'name' => $command->getName(),
             'usage' => array_merge(array($command->getSynopsis()), $command->getUsages(), $command->getAliases()),
             'description' => $command->getDescription(),
             'help' => $command->getProcessedHelp(),
-            'definition' => $this->getInputDefinitionData($command->getNativeDefinition()),
+            'definition' => $this->getInputDefinitionData($command->getDefinition()),
         );
```

Comment:
Are the changes on the descriptors really needed ? I don't see the benefit of using `getDefinition` instead of `getNativeDefinition` ? Furthermore doesn't it introduce a BC break when `getNativeDefinition` is overridden ?

Reply:
`getDefinition()` is the actual command definition, potentially merged with its application definition. IMHO we should describe it fully, as these options/args actually work.

Reply:
Indeed, the previous behavior is kept. It remains the case where a developper override `getNativeDefinition` with something other than the non-merged definition. But maybe it can be considered as an incorrect usage of `getNativeDefinition`. I don't know if it's a "real" BC break.

Reply:
fixed in 5f4082860173c3086aa86687f4c211308f79d15f